### PR TITLE
Set only the body of methods when we are in generation

### PR DESCRIPTION
**Pull request by goldhoorn**

### CODE REVIEW

**Comment by goldhoorn** [comment]
This PR superseeds #40. The Problem is that the body can be left unknown for the spec. Nevertheless to keep backward compatibility we allow the function call on the task but ignore the body if we are not during generation phase.

The other solution would be to remove the attribute extension from the RTT::CPP part and save the body inside the spec. (What would not make sense either).

**Comment by doudou** [comment]
Hi. While I like the intent, I would appreciate it if you could make it more complete. As you notice yourself, it makes no sense to have body specification in the spec part. We should therefore get rid of it, that is:

 - add a deprecation warning to the body argument in hidden_operation (and check how plain operations work)
 - make #base_body and #body DSL setters (using dsl_attribute) in the operation part
 - force people to call them in register_for_generation

**Comment by goldhoorn** [comment]
don't have the time (now), if someone else would like to take over ping me otherwise it's on my todo
@christian-rauch ?

**Comment by goldhoorn** [comment]
ok had found a few mins *updated*

**Comment by goldhoorn** [comment]
*updated pr* i added a second call to usercpase plugins which allows to set/modify already created properties. Otherwise operation bodys cannot be setted/modified before the method generation.

File: lib/orogen/spec/task_context.rb [hunk 1]
```diff
@@ -758,10 +758,13 @@ def operation(name)
 
             # Defines an operation whose implementation is in the Base class
             # (i.e. "hidden" from the user)
-            def hidden_operation(name, body)
+            def hidden_operation(name, body=nil)
                 op = operation(name)
                 op.hidden = true
-                op.base_body = body
+                if(body)
```
Comment (doudou) [inline lib/orogen/spec/task_context.rb]: Please don't put () around if arguments ...

File: lib/orogen/spec/task_context.rb [hunk 2]
```diff
@@ -758,10 +758,13 @@ def operation(name)
...
+                    OroGen.warn "body argument for hidden_operation '#{self.name}'.'#{name}' is deprecated, please set the body during generation phase"
```
Comment (doudou) [inline lib/orogen/spec/task_context.rb]: Mention the early_... method ?

**Comment by goldhoorn** [comment]
*updated*

File: lib/orogen/gen/tasks.rb [hunk 3]
```diff
@@ -175,6 +175,10 @@ def register_for_generation
         # Module that is used to add code generation functionality to
         # Spec::Operation
         module OperationGeneration
+
+            dsl_attribute(:body) { |value| value.to_s }
```
Comment (doudou) [inline lib/orogen/gen/tasks.rb]: Indentation's fucked up

**Comment by goldhoorn** [comment]
no clue how the tab was sneaked in but *fixed*

**Pull request merged**